analyses/new-feature-template.qmd

```diff
@@ -0,0 +1,1549 @@
+
+
+  datatable(category_percentages_township,
+    options = list(
+      scrollY = "300px",
+      scrollX = TRUE,
+      paging = FALSE,
+      searching = TRUE
+    ),
+    rownames = FALSE
+  )
+}
+```
+
+## Neighborhood
+
+```{r mean_median_neighborhood}
+if (params$type == "continuous") {
+  descriptives_nbhd <- working_data_pin %>%
+    group_by(meta_nbhd_code) %>%
+    summarize(
+      mean = round(mean(!!sym(params$added_feature), na.rm = TRUE), 2),
+      median = round(median(!!sym(params$added_feature), na.rm = TRUE), 2)
+    )
+
+  datatable(descriptives_nbhd,
+    options = list(
+      scrollY = "300px",
+      scrollX = TRUE,
+      paging = FALSE,
+      searching = TRUE
+    ),
+    rownames = FALSE
+  )
+} else if (params$type == "categorical") {
+  category_percentages_nbhd <- working_data_pin %>%
+    group_by(meta_nbhd_code, !!sym(params$added_feature)) %>%
+    count() %>%
+    mutate(percentage = n / sum(n) * 100) %>%
+    select(meta_nbhd_code, !!sym(params$added_feature), percentage)
+
+  datatable(category_percentages_nbhd,
+    options = list(
+      scrollY = "300px",
+      scrollX = TRUE,
+      paging = FALSE,
+      searching = TRUE
+    ),
+    rownames = FALSE
+  )
+}
+```
+
+
+:::
+
+## Histogram
+
+```{r histogram}
+if (params$type == "continuous") {
+  working_data_pin %>%
+    mutate(
+      mean_value = mean(!!sym(params$added_feature), na.rm = TRUE),
+      median_value = median(!!sym(params$added_feature), na.rm = TRUE)
+    ) %>%
+    ggplot(aes(x = !!sym(params$added_feature))) +
+    geom_histogram(fill = "blue", color = "black", alpha = 0.7) +
+    geom_vline(aes(xintercept = mean_value, color = "Mean"), linetype = "dashed", linewidth = 1, show.legend = TRUE) +
+    geom_vline(aes(xintercept = median_value, color = "Median"), linetype = "dashed", linewidth = 1, show.legend = TRUE) +
+    scale_color_manual(
+      name = "Statistics",
+      values = c(Mean = "red", Median = "green"),
+      labels = c("Mean", "Median")
+    ) +
+    labs(
+      x = params$added_feature,
+      y = "Frequency"
+    ) +
+    theme_minimal()
+} else if (params$type == "categorical") {
+  category_percentages <- working_data_pin %>%
+    count(!!sym(params$added_feature)) %>%
+    mutate(percentage = n / sum(n) * 100)
+
+  ggplot(category_percentages, aes(x = !!sym(params$added_feature), y = percentage)) +
+    geom_bar(stat = "identity", fill = "blue", color = "black", alpha = 0.7) +
+    labs(
+      x = params$added_feature,
+      y = "Percentage"
+    ) +
+    theme_minimal()
+}
+```
+
+
+## FMV Change Histogram
+
+This chart shows the distribution of the value of 'diff_pred_pin_initial_fmv' in the model with the added feature minus the model without the added feature. Outliers outside of 95% are removed to make the chart more readable. The largest 100 increases and decreases are displayed in maps in section X. 
```

Comment:
A good number of charts are bounded within 95%. For this dataset, there are characteristics which go to 30,000 feet which really skew the charts.